File: BACKEND/Tutorial/src/ApplicationCore/Entities/Framework/Attachment.cs
namespace Tutorial.ApplicationCore.Entities
{
	public class Attachment : BaseEntity
	{
		public int Id { get; set; }
		public string OriginalFileName { get; set; }
		public string SavedFileName { get; set; }		
		public string FileExtension { get; set; }
		public long FileSize { get; set; }

		private string downloadUrl = "";
		public string DownloadURL
		{
			get
			{
				return downloadUrl;
			}
		}

		public void ComposeDownloadUrl(string composedUri)
		{
			downloadUrl = composedUri;
		}

		public Attachment()
		{

		}

		public Attachment(string originalFilename, string savedFilename, string fileExtension, long fileSize)
		{
			OriginalFileName = originalFilename;
			SavedFileName = savedFilename;
			FileExtension = fileExtension;
			FileSize = fileSize;
		}
	}
}

File: FRONTEND/src/models/Core/Attachment.js
import BaseModel from '../BaseModel';

export default class Attachment extends BaseModel {
  static entity = 'attachment'
  static entityName = "attachment";
  
  static fields () {
    return {
      id : this.uid(),
      downloadUrl : this.string(null).nullable(),
      originalFilename : this.string(null).nullable(),
      savedFileName : this.string(null).nullable(),
      fileExtension : this.string(null).nullable(),
      fileSize : this.number(0).nullable(),
    }
  }
}
